perun/collect/trace/run.py

```diff
@@ -1,206 +1,275 @@
-    kwargs.setdefault('cleanup', True)
-    kwargs.setdefault('verbose_trace', False)
+    # No runnable command was given, terminate the collection
+    if not kwargs['binary'] and not executable.cmd:
+        raise InvalidBinaryException('')
+    # Otherwise copy the cmd or binary parameter
+    elif not executable.cmd:
+        executable.cmd = kwargs['binary']
+    elif not kwargs['binary']:
+        kwargs['binary'] = executable.cmd
 
-    # Set the binary to cmd if not provided and check that it exists
-    if not kwargs['binary']:
-        kwargs['binary'] = kwargs['executable'].cmd
+    # Check that the binary / executable file exists and is valid
     kwargs['binary'] = os.path.realpath(kwargs['binary'])
     if not os.path.exists(kwargs['binary']) or not utils.is_executable_elf(kwargs['binary']):
-        raise exceptions.InvalidBinaryException(kwargs['binary'])
+        raise InvalidBinaryException(kwargs['binary'])
+    elif not os.path.exists(executable.cmd):
+        raise InvalidBinaryException(executable.cmd)
+
+    # Update the configuration dictionary with some additional values
+    kwargs['executable'] = executable
+    kwargs['timestamp'] = time.strftime("%Y-%m-%d-%H-%M-%S", time.localtime())
+    kwargs['pid'] = os.getpid()
+    kwargs['files_dir'] = temp.temp_path(os.path.join('trace', 'files'))
+    kwargs['locks_dir'] = temp.temp_path(os.path.join('trace', 'locks'))
+    # Init the resources object that is used for a clean teardown and store it in the resources list
+    # This makes the resources available even if 'before' fails and kwargs is not updated
+    kwargs['opened_resources'].append(Res())
+    kwargs['res'] = kwargs['opened_resources'][0]  # Alias for easier access to the resource object
     return kwargs
 
 
-def _create_collector_file(name, suffix='.txt', **kwargs):
-    """Prepares path to the requested file used during collection.
-    The format is as follows: 'collect_<name>_<timestamp><suffix>
+def _create_collect_files(timestamp, pid, res, output_handling, **kwargs):
+    """ Creates the temporary files required for data collection. Namely:
+      - the SystemTap script file
+      - the SystemTap log file
+      - the SystemTap data file (i.e. where the raw measured performance data are stored)
+      - the output capture file (optional) used to store stdout / stderr of the profiled command
 
-    :param str name: further specification of collector file name
-    :param str suffix: the file suffix
-    :param kwargs: additional parameters, e.g. timestamp
-    :return str: assembled path to the file
+    :param str timestamp: the perun startup timestamp
+    :param int pid: the PID of the running perun process
+    :param Res res: the resources object
+    :param OutputHandling output_handling: determines whether the output capture file is required
+    :param kwargs: additional configuration values
+    """
+    files = [(Res.script(), '.stp'), (Res.log(), '.txt'), (Res.data(), '.txt')]
```

Comment:
Suggestion: Consider renaming Res to something longer.

Reply:
The Res class will be removed in the new tracer version.